### PR TITLE
[AIRFLOW-XXXX] document system tests mechanism better

### PR DESCRIPTION
**Pull request by potiuk**
---
Issue link: `Document only change, no JIRA issue`

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: TESTING.rst [hunk 1]
```diff
@@ -543,31 +587,197 @@ tests whenever an operator/hook/sensor is added/modified in a given system.
...
+
+
+Iteration with System Tests if your resources are slow to create
+----------------------------------------------------------------
+
+When you want to iterate on system tests, you might want to create slow resources first
+
+If you need to setup some external resources for your tests (for example compute instances in Google Cloud)
+you should set them up and teardown in the setUp/tearDown methods of your tests.
+Since those resources might be slow to create you might want to add some helpers that
```
Comment (michalslowikowski00) [inline TESTING.rst:701]: Too long sentence, imo.

File: TESTING.rst [hunk 2]
```diff
@@ -486,16 +486,57 @@ The system tests execute a specified example DAG file that runs the DAG end-to-e
...
---------------------
+Environment for System Tests
+----------------------------
+
 **Prerequisites:** You may need to set some variables to run system tests. If you need to
-add some intialization of environment variables to Breeze, you can always add a
+add some initialization of environment variables to Breeze, you can add a
 ``variables.env`` file in the ``files/airflow-breeze-config/variables.env`` file. It will be automatically
-sourced when entering the Breeze environment.
+sourced when entering the Breeze environment. You can also add some additional initialization commands
```
Comment (michalslowikowski00) [inline TESTING.rst]: Where those additional commands can be added? In `files/airflow-breeze-config/variables.env`

File: TESTING.rst [hunk 3]
```diff
@@ -543,31 +587,197 @@ tests whenever an operator/hook/sensor is added/modified in a given system.
...
+
+.. code-block:: bash
+
+   pytest -o faulthandler_timeout=2400 --systems=google tests/providers/google/cloud/operators/test_compute_system.py
+
+
+Iteration with System Tests if your resources are slow to create
+----------------------------------------------------------------
+
+When you want to iterate on system tests, you might want to create slow resources first
```
Comment (michalslowikowski00) [inline TESTING.rst]: When you want to iterate on system tests, you might want to create slow resources first.

Missing dot at the end.

**Review by turbaszek — APPROVED** [review]

File: TESTING.rst [hunk 4]
```diff
@@ -543,31 +587,197 @@ tests whenever an operator/hook/sensor is added/modified in a given system.
...
+to create and set-up so that you can iterate on running the system tests without
+loosing the time for creating theme every time. A temporary file is created to prevent from
+setting up and tearing down the instances when running the test.
+
+This example also shows how you can use the random number generated at the entry of Breeze if you
+have it in your variables.env (see previous chapter). In case of Cloud SQL you cannot reuse the
+same instance name for a week so we generate random number that is used across the whole session
+and store it in ``/random.txt`` file so that the names are unique during tests.
+
+NOTE! Do not forget to delete manually created resources before leaving the Breeze session. They
```
Comment (michalslowikowski00) [inline TESTING.rst]: Can you mark it red or highlighted to make it more visible. WDYT?

File: TESTING.rst [hunk 5]
```diff
@@ -543,31 +587,197 @@ tests whenever an operator/hook/sensor is added/modified in a given system.
...
+   every time you change the operators/hooks or example_dags. The example below shows reinstallation
+   of the google package:
+
+In the host:
+
+.. code-block:: bash
+
+  ./scripts/ci/ci_prepare_backport_packages.sh google
+
+In the container (:
```
Comment (michalslowikowski00) [inline TESTING.rst]: Typo? `(:`

**Review by michalslowikowski00 — COMMENTED** [review]
Nice! 👍

File: TESTING.rst (hunk 4 above)
Comment (potiuk) [reply TESTING.rst]: Highlighted!

**Comment by potiuk** [comment]
Thanks fixed all points :)

**Pull request merged**